### ISSUE TITLE
Did anyone managed to get imessage and facetime working on Big Sur ? 

### ISSUE THREAD

**Issue by Lels07** [daliansky/XiaoMi-Pro-Hackintosh#538]
I read https://www.tonymacx86.com/threads/an-idiots-guide-to-imessage.196827/  and https://www.tonymacx86.com/threads/how-to-fix-imessage.110471/, even with this i failed to get imessage and facetime working on my laptop TM1707. No error message, just iMessage asking me again for login after 20sec.

**Comment by Menchen** [comment]
Try using a new serial number after deleting all cache file

**Comment by Lels07** [comment]
i have already try, and it still not working, i'm curious to know if someone succeed to get Imessage or facetime working on Big Sur

**Comment by adeniji57** [comment]
@Lels07 I had this problem a while ago and I eventually found out that Apple had flagged my account to stop it from being able to log in. All I had to do was call their Customer Support.

**Comment by Lels07** [comment]
@adeniji57 Yes, i found that had this problem to, but i tried with another account (Valid) and the problem still persist, i made a call last week and they told me that my ID was not flagged or placed in SPAM

**Comment by SimonCaignart** [comment]
> I read https://www.tonymacx86.com/threads/an-idiots-guide-to-imessage.196827/ and https://www.tonymacx86.com/threads/how-to-fix-imessage.110471/, even with this i failed to get imessage and facetime working on my laptop TM1707. No error message, just iMessage asking me again for login after 20sec.

Same for me, you are not the only one

**Comment by SimonCaignart** [comment]
@Lels07 have you tried to follow this guide ? [https://dortania.github.io/OpenCore-Post-Install/universal/iservices.html#generate-a-new-serial](url)

I can't manage to have my network interface to en0, I tried to delete the two files but It is always en1, and en2 if I connect through ethernet.

Adding the NullEthernet.kext and the ssd-rmne.aml somehow broke the hardware acceleration ( no more transparency and os not fluid ), but I could have made mistakes.

You can try the solutions from the link, make sure to backup your EFI folder before doing it so you can rollback if you have the same problem ( hardware acceleration ).

I will be happy to know if you solved the problem so I can too. :)

**Comment by Lels07** [comment]
@SimonCaignart, i have tried this guide,  nullEthernet.kext and ssd-rmne are already in the release-EFI of this repo. It simulate a built-in Ethernet port with bsdname en0 and it seems works fine on the Notebook Pro. I also use AirportItlw to drive native intel Wireless card, so i dont use a USB Wifi Dongle so according to https://github.com/RehabMan/OS-X-Null-Ethernet i think i dont need the NullEthernet.kext. if i remove NullEthernet, my native intel wireless card appears with bsdname en0 and everything working fine except Imessage and Facetime.    
@stevezhengshiqi is NullEthernet.kext useful even with AirportItlw.kext ? 
![Capture d’écran 2021-02-17 à 14 24 56](https://user-images.githubusercontent.com/62592908/108210835-52685100-712c-11eb-84af-47670e5a49c9.png)

**Comment by stevezhengshiqi** [comment]
If you have `Airportitlwm` working, `NullEthernet` is not necessary. It's a failsafe when `Airportitlwm` does not load, which happens occasionally. With NullEtherenet, it's recommended to change MAC address in [SSDT-RMNE](https://github.com/daliansky/XiaoMi-Pro-Hackintosh/blob/main/ACPI/Shared/Source/SSDT-RMNE.dsl#L13). The default 112233445566 is on Apple's blacklist.

**Comment by Lels07** [comment]
@stevezhengshiqi thanks!
clearly, this is not a SMBIOS configuration problem or serial number. i have no error message, just imessage sign out 5 seconds after sign in. Maybe it's relative to the apple ID and a possible flagged ID but i tried with another account after deleting caches files and change serial number and the problem persist. This issue seems quiet odd, maybe i can try a fresh install with a new account

**Comment by Lels07** [comment]
The problem is solved by using an old apple ID created few years ago. Every new Id that i created not working on my hackintosh with imessage and facetime.

**Comment by orhan-drsn** [comment]
> The problem is solved by using an old apple ID created few years ago. Every new Id that i created not working on my hackintosh with imessage and facetime.

But how can I solve this Problem as a new user

**Comment by SimonCaignart** [comment]
> The problem is solved by using an old apple ID created few years ago. Every new Id that i created not working on my hackintosh with imessage and facetime.

Effectively, using an older account worked. After I had added a good serial, good rom in config.plist and also in SSDT-RMNE, I had no more ideas than trying to connect with another account, so I asked a friend to connect with her account on my laptop, and iMessage and FaceTime worked perfectly, she also has an iPhone, maybe that's why with her account it is working.
But of course that's not an acceptable fix because I can't use her account for everyday use haha.
Maybe mine is banned somehow ? but if you already tried creating new ones and it kept not working, we are in trouble.
Something we could try is to link our account to another apple product, like an iPad or iPhone, to see if it helps to "unban" our account. I will try that with an iPad when I have the chances to get my hands on it.

**Comment by mBilG** [comment]
I tried with one new account and one from a few years back.. It has nothing to do with using on iPad or iPhone as both accounts are linked to an iPad.. but both still failed.. not sure if the "old" account is not old enough?
I even tried contacting Apple for the account.. they ran me through all the steps that we already know online to reset NVRAM and keychain etc.. In the end I had to end the call because of work.. If anyone is able to get Apple to unblock their ID let us know.. because i see even real mac users also face the same issue on their macs.. it's probably a protocol that Apple has set..

**Comment by SimonCaignart** [comment]
> I tried with one new account and one from a few years back.. It has nothing to do with using on iPad or iPhone as both accounts are linked to an iPad.. but both still failed.. not sure if the "old" account is not old enough?
> I even tried contacting Apple for the account.. they ran me through all the steps that we already know online to reset NVRAM and keychain etc.. In the end I had to end the call because of work.. If anyone is able to get Apple to unblock their ID let us know.. because i see even real mac users also face the same issue on their macs.. it's probably a protocol that Apple has set..

Mmmmh, maybe an iPad is not enough, i think i will try to link an iPhone, with a real phone and a number linked to the account and to iMessage it could work maybe.

**Comment by mBilG** [comment]
> Mmmmh, maybe an iPad is not enough, i think i will try to link an iPhone, with a real phone and a number linked to the account and to iMessage it could work maybe.

Maybe that could work..  do update us once you have anything..

**Comment by Lels07** [comment]
The old account that i use to "fix" (not really a fix in fact i agree) the problem was not link to an iphone so i dont think this is the problem

**Comment by SimonCaignart** [comment]
> The old account that i use to "fix" (not really a fix in fact i agree) the problem was not link to an iphone so i dont think this is the problem

Oh ok, I will not waste my time trying to link my account with an iPhone ahah, thanks

By Searching the web I found out that it's a pretty wide problem, also affecting the real Mac users 
https://discussions.apple.com/thread/252032274
https://www.tonymacx86.com/threads/unable-to-login-to-imessage-on-big-sur.303322/

And from what i see in the comments, calling apple is maybe the right move, our accounts must be banned server side. Telling them that we already tried resetting nvram ect, and that with another account it works, they will hopefully tell us to restart and unban us.

**Comment by SimonCaignart** [comment]
Little update, I called apple support today, during a 30 mn + call with 2 people, I did not manage to get useful help, I explained that it was probably a server side ban of my account because I had tried with another one before and it worked, but they confirmed me several times that my account was not blocked server side by apple. The only solution given to me was to reinstall the system. So ... I have no more ideas :/

**Comment by mBilG** [comment]
> Little update, I called apple support today, during a 30 mn + call with 2 people, I did not manage to get useful help, I explained that it was probably a server side ban of my account because I had tried with another one before and it worked, but they confirmed me several times that my account was not blocked server side by apple. The only solution given to me was to reinstall the system. So ... I have no more ideas :/

That's unfortunate.. They're probably filtering the Serial Numbers then..

**Comment by mBilG** [comment]
@stevezhengshiqi quick question, is it possible to change the "board-id" in OC or something?
From IORegistryExplorer, under Root > / > efi > platform > board-id, the value is "Mac-5F9802EFE386AA28".. however, [this](https://mrmacintosh.com/list-of-mac-boardid-deviceid-model-identifiers-machine-models/) shows that for this SMBIOS, the board-id should be Mac-E7203C0F68AA0004.. 
They also mention that starting from Big Sur, Apple is checking this and the T2 ID.. could that possibly be why iMessage isn't working?

**Comment by stevezhengshiqi** [comment]
@mBilG board-id is not related to imessage/facetime. The ultimate solution is using a real Mac's smUUID, MLB and serialnumber.

**Comment by mBilG** [comment]
> @mBilG board-id is not related to imessage/facetime. The ultimate solution is using a real Mac's smUUID, MLB and serialnumber.

Alright.. but does this value serve any real purpose on hackintosh? or is it just cosmetic?

**Comment by stevezhengshiqi** [comment]
board-id is bond to the SMBIOS model. macOS will execute the corresponding AGPM & X86PluginFamily profile to manage the power management and performance.

**Comment by Knives79** [comment]
I have 2 PC, this notebook and a Ryzen Desktop. On the desktop I have no issue at all and iMessage+Facetime works flawlessly. On this notebook I was never able to make it work. I will try to copy the serials from the desktop, maybe it will work...

**Comment by Menchen** [comment]
@Knives79 Follow https://dortania.github.io/OpenCore-Post-Install/universal/iservices.html to generate serial, note that you should replace the device model from your config, copying from another machine is a bad idea as you risk apple banning the serial, also the device type won’t match (mac vs macbook). Clean the nvram and remove old file before rebooting with new serial.

**Comment by Knives79** [comment]
Wow! I just installed the Sonoma update: now everything is working without changing anything! That's cool!

**Comment by Knyyl** [comment]
Yea both work for me
